### ISSUE TITLE
Table of Contents links and Permalink IDs don't match

### ISSUE THREAD

**Issue by onigoetz** [thephpleague/commonmark#690]
**Version(s) affected**: 2.0.0

**Description**  
When using `HeadingPermalinkExtension` along with `TableOfContentsExtension` links in the table of contents won't link to the corresponding Permalink

**How to reproduce**  

```php
use League\CommonMark\Environment\Environment;
use League\CommonMark\Extension\CommonMark\CommonMarkCoreExtension;
use League\CommonMark\Extension\HeadingPermalink\HeadingPermalinkExtension;
use League\CommonMark\Extension\TableOfContents\TableOfContentsExtension;
use League\CommonMark\MarkdownConverter;

$environment = new Environment([]);
$environment->addExtension(new CommonMarkCoreExtension());
$environment->addExtension(new HeadingPermalinkExtension());
$environment->addExtension(new TableOfContentsExtension());

$converter = new MarkdownConverter($environment);

echo $converter->convertToHtml("# Test");
```

Produces

```html
<ul class="table-of-contents">
  <li>
    <a href="#test">Test</a>
  </li>
</ul>
<h1>
  <a id="content-test" href="#content-test" class="heading-permalink" aria-hidden="true" title="Permalink">¶</a>
  Test
</h1>
```

As you can see `href="#test"` doesn't match with `id="content-test"`

**Comment by onigoetz** [comment]
Oh and the [Table of Contents extension documentation page](https://commonmark.thephpleague.com/2.0/extensions/table-of-contents/) isn't listed on the left sidebar with the other extensions. is this on purpose ?

**Comment by onigoetz** [comment]
I did some digging and it seems the `HeadingPermalinkRenderer` adds the prefix to the `id` at the very last moment : https://github.com/thephpleague/commonmark/blob/main/src/Extension/HeadingPermalink/HeadingPermalinkRenderer.php#L63

But the `TableOfContentsGenerator` doesn't do anything with those prefixes: https://github.com/thephpleague/commonmark/blob/main/src/Extension/TableOfContents/TableOfContentsGenerator.php#L90

**Comment by colinodell** [comment]
The missing menu link is definitely an accident - thanks for letting me know!

The reason for having separate fragment and ID prefixes is due to #664 where we changed permalinks to create and target `id` attributes instead of using `name`.  Some developers may want to prefix the `id` attributes to prevent users from accidentally or intentionally using IDs that collide with other elements on the page, especially given that CSS or JS might be targeting those IDs.  
 This helps avoid potential security and design issues.  But developers might want to create pretty URLs that exclude that prefix and use some JS magic to "rewrite" those fragment links.  Most users of this library probably won't care enough to use separate prefixes, especially if only trusted users are creating Markdown, but I wanted to provide flexibility out-of-the-box for those who want it.

(GitHub does something similar with their autolinks - the IDs are prefixed with `user-content-` but the URL fragments exlude this prefix.)

I have a fix prepared for this bug but I'll need to update some tests too, and it's getting late, so I'll wrap this up and release a fix later this week :)

**Comment by onigoetz** [comment]
Thanks for the update, I didn't know of that use case with GitHub, interesting :)

I'm in no hurry, take your time :)

**Comment by colinodell** [comment]
Fixed via https://github.com/thephpleague/commonmark/commit/1cc648c1808c9989fb97ce16f66dca9cf0d4ab61

**Comment by colinodell** [comment]
Fix has been released in 2.0.1! :tada:
